src/v/pandaproxy/kafka_client_cache.cc

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "pandaproxy/kafka_client_cache.h"
+
+namespace pandaproxy {
+
+kafka_client_cache::kafka_client_cache(
+  YAML::Node const& cfg,
```

Comment:
> taking the yaml node seems like it is too low level. is there a reason for that over a configuration binding, one or more static values, or accessing the thread_local config() object directly in the implementation?

It's a little unfortunate, but the configuration requires a copy, which isn't supported by `config::property`. Once upon a time I had a go at creating a polymorphic clone method, but I forget where I got stuck.

Reply:
> > taking the yaml node seems like it is too low level. is there a reason for that over a configuration binding, one or more static values, or accessing the thread_local config() object directly in the implementation?
> 
> It's a little unfortunate, but the configuration requires a copy, which isn't supported by `config::property`. Once upon a time I had a go at creating a polymorphic clone method, but I forget where I got stuck.

And changing property seems dangerous in my head.

Reply:
are you snagging static values out the config? perhaps it makes sense to pass in a `struct cache_config` object and set the values. passing in a YAML::Node& makes testing awkward.

Reply:
It might be worth deferring this change. The constructor for `kafka::client` take a `YAML::Node`, this parameter is just plumbing.

Reply:
> It might be worth deferring this change. The constructor for `kafka::client` take a `YAML::Node`, this parameter is just plumbing.

Yeah I should've said that first, the client only needs a YAML::Node.
I can, however, probably access the thread local config object directly.